### ISSUE TITLE
`std` uses a `Thin` trait alias which breaks mir eval

### ISSUE THREAD

**Issue by Veykril** [rust-lang/rust-analyzer#15566]
https://doc.rust-lang.org/std/ptr/fn.null_mut.html is defined as:
```rs
pub const fn null_mut<T: ?Sized + Thin>() -> *mut T;
```
where `Thin` is defined as:
```rs
pub trait Thin = Pointee<Metadata = ()>;
```

r-a currently does not handle trait aliases correctly, as such mir eval bails out on almost all code that does `Vec` allocations via `Vec::with_capacity` because we lack the knowledge that `Metadata = ()` in `null_mut` resulting in a `Expected <T as Pointee>::Metadata, found ()` error.

**Comment by Veykril** [comment]
@HKalbasi this is the cause for the mir failures (or at least part of them)

**Comment by HKalbasi** [comment]
Yes it seems this is the reason of failure in mir eval of debug impls. For mir lowering failures the problem was in the stats, fixed by #15567
